src/client-side-encryption/client_encryption.ts

```diff
@@ -213,32 +191,17 @@ export class ClientEncryption implements StateMachineExecutable {
       tlsOptions: this._tlsOptions
     });
 
-    // @ts-expect-error We did not convert promiseOrCallback to TS
-    return promiseOrCallback(callback, cb => {
-      stateMachine.execute<DataKey>(this, context, (err, dataKey) => {
-        if (err || !dataKey) {
-          cb(err, null);
-          return;
-        }
+    const updatedDataKey = await stateMachine.executeAsync<DataKey>(this, context);
```

Comment:
I have to choose another name here because we were shadowing. Up for any suggestions

Reply:
maybe just remove the first declaration and name this `dataKey`?  it's not proving much value imo and it actually makes this more confusing, because the data key is created by `stateMachine.execute`.

```typescript
    const dataKeyBson = serialize({ provider, masterKey: options.masterKey });
```